community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ParallelBatchImporter.java

```diff
@@ -226,6 +226,13 @@ public void doImport( Input input ) throws IOException
         }
     }
 
+    private File badRelationshipsFileName()
+    {
+        String badFileName = config.badFileName();
+        File badFile = new File( badFileName );
+        return badFile.isAbsolute() ? badFile : new File( storeDir, badFileName );
+    }
+
```

Comment:
Why don't you move this computation into the implementation of `Configuration`? That'd fix my comment above.